lib/utils/flattenNestedSelectorsForRule.mjs

```diff
@@ -0,0 +1,43 @@
+import resolvedNestedSelector from 'postcss-resolve-nested-selector';
+
+import getRuleSelector from './getRuleSelector.mjs';
+import getSelectorAST from './getSelectorAST.mjs';
+import isStandardSyntaxSelector from './isStandardSyntaxSelector.mjs';
+
+/**
+ * @typedef {import('postcss-selector-parser').Selector} Selector
+ * @typedef {import('postcss-selector-parser').Root} SelectorRoot
+ * @typedef {Array<{selector: Selector, resolvedSelectors: Array<Selector>}> | undefined} FlattenedSelectors
+ * @param {import('postcss').Rule} rule
+ * @param {import('stylelint').PostcssResult} result
+ * @returns {Array<{selector: Selector, resolvedSelectors: Array<Selector>}> | undefined}
+ */
+export default function flattenNestedSelectorsForRule(rule, result) {
+	/** @type {FlattenedSelectors} */
+	const flattenedSelectors = [];
+
+	const ownAST = getSelectorAST(getRuleSelector(rule), result, rule);
+
+	if (!ownAST) return;
+
+	for (const selectorAST of ownAST.nodes) {
+		const resolvedSelectors = resolvedNestedSelector(selectorAST.toString(), rule);
+
+		for (const resolvedSelector of resolvedSelectors) {
+			if (!isStandardSyntaxSelector(resolvedSelector)) return;
```

Comment:
[ask] Just to confirm, does not skipping a non-standard selector make sense?

```suggestion
			if (!isStandardSyntaxSelector(resolvedSelector)) continue;
```

Reply:
I would prefer to keep it as it currently is and skip the entire selector list.

I am concerned that the updated implementation using `@csstools/selector-resolve-nested` will not be able to handle partially non-standard selectors because it always first parses and then processes selectors.

Reply:
Make sense. Let's keep it as-is 👍🏼